e2e_tests/integrations/test_huggingface.py

```diff
@@ -461,3 +463,173 @@ def test_usages(self):
         assert "NeptuneCallback" in [
             type(callback).__name__ for callback in trainer.callback_handler.callbacks
         ]
+
+    def test_model_checkpoints_same(self, environment):
+        with init_run(project=environment.project, api_token=environment.user_token) as run:
+            run_id = run["sys/id"].fetch()
+            callback = NeptuneCallback(run=run, log_checkpoints="same")
+            training_args = self._trainer_default_attributes.copy()
+            training_args["args"] = TrainingArguments(
+                "model", report_to=[], num_train_epochs=500, save_steps=1000, save_strategy="steps"
+            )
+            trainer = Trainer(**training_args, callbacks=[callback])
+            trainer.train()
+
+        run = init_run(
+            run=run_id,
+            project=environment.project,
+            api_token=environment.user_token,
+            mode="read-only",
+        )
+
+        assert run.exists("finetuning/checkpoints")
+        assert run["finetuning/train/epoch"].fetch_last() == 500
+        with tmp_context():
+            run["finetuning/checkpoints"].download("checkpoints.zip")
+
+            with ZipFile("checkpoints.zip") as handler:
+                assert set([os.path.dirname(x) for x in handler.namelist()]) == {
+                    "/",
+                    "model",
+                    "model/checkpoint-1000",
+                    "model/checkpoint-2000",
+                }
+                handler.extractall(".")
+
+        with init_run(project=environment.project, api_token=environment.user_token) as run:
+            run_id = run["sys/id"].fetch()
+            callback = NeptuneCallback(run=run)
+            training_args = self._trainer_default_attributes.copy()
+            training_args["args"] = TrainingArguments("model", report_to=[], num_train_epochs=1000)
+            trainer = Trainer(**training_args, callbacks=[callback])
+            trainer.train(resume_from_checkpoint="model/checkpoint-2000")
+
+        run = init_run(
+            run=run_id,
+            project=environment.project,
+            api_token=environment.user_token,
+            mode="read-only",
+        )
+        assert run["finetuning/train/epoch"].fetch_last() == 1000
```

Comment:
I think that we should use `_test_with_run_initialization` twice in this test.

Reply:
The same is true for other tests.

Reply:
Done